### PR TITLE
Kraken 0.10.5-beta (FIX: missing jellyfish-1.1 keg deps)

### PR DESCRIPTION
**Pull request by tseemann**
Had to patch in old version of jellyfish.
This only affects people who want to build their _own_ kmer databases (like me).
Searching against the databases available at the Kraken website works ok without this.

Also, install docs too.

### CODE REVIEW

File: kraken.rb [hunk 1]
```diff
@@ -13,15 +13,24 @@ class Kraken < Formula
 
   needs :openmp
 
+  depends_on 'jellyfish-1.1'  # this is keg_only
+
   def install
+    # ensure we refer to jellyfish 1.x keg and not the incompatible jellyfish (2.x) package
+    jf1 = Formula["jellyfish-1.1"].bin / "jellyfish"
```
Comment (iml) [inline kraken.rb]: Use opt_bin when referring to external formulae. Doesn't matter if it's keg_only. Also prefer not to have spaces in between strings like that.

**Comment by iml** [comment]
```
==> brew style kraken
== /Library/Caches/Homebrew/Formula/kraken.rb ==
C: 16: 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 34:  6: Inconsistent indentation detected.
C: 34: 54: Space inside parentheses detected.
C: 34: 87: Space inside parentheses detected.

1 file inspected, 4 offenses detected

**Comment by tseemann** [comment]
@iml Thanks for tip on 'brew style'. I've made the fixes.

**Comment by iml** [comment]
@BrewTestBot test this please.

**Comment by iml** [comment]
@BrewTestBot test this please.

**Comment by tseemann** [comment]
@BrewTestBot test this please.

**Comment by tseemann** [comment]
```
       Testing homebrew/science/kraken
==> /usr/local/Cellar/kraken/0.10.5-beta/bin/kraken --version
Kraken version 0.10.5-beta
Copyright 2013-2015, Derrick Wood (dwood@cs.jhu.edu)
==> /usr/local/Cellar/kraken/0.10.5-beta/libexec/check_for_jellyfish.sh
dyld: Symbol not found: __ZNSt8__detail15_List_node_base7_M_hookEPS0_
  Referenced from: /usr/local/opt/jellyfish-1.1/bin/jellyfish
  Expected in: /usr/lib/libstdc++.6.dylib
 in /usr/local/opt/jellyfish-1.1/bin/jellyfish
Error: homebrew/science/kraken: failed
<0> expected but was
<133>.
```

**Comment by karel-brinda** [comment]
@tseemann Is there any possibility how to install Kraken + JellyFish 1.1 with ``brew`` to make it work?

**Comment by tseemann** [comment]
@karel-brinda That is the whole point of this pull request. It allows jellyfish1 to be installed as a Keg (not in the PATH) and it makes sure Kraken uses that one and not the one in the PATH. 

I'm not sure why it isn't working yet, I need to check it.

**Comment by tseemann** [comment]
This is some sort of OSX C library issue or something?

**Comment by tseemann** [comment]
Fails on Yos and Mav:
Is this something to do with `jellyfish-1.1` being `keg_only` ?

```
==> /usr/local/Cellar/kraken/0.10.5-beta/libexec/check_for_jellyfish.sh
dyld: Symbol not found: __ZNSt8__detail15_List_node_base7_M_hookEPS0_
  Referenced from: /usr/local/opt/jellyfish-1.1/bin/jellyfish
  Expected in: /usr/lib/libstdc++.6.dylib
 in /usr/local/opt/jellyfish-1.1/bin/jellyfish
Error: homebrew/science/kraken: failed
<0> expected but was
<133>.
```

**Comment by ilovezfs** [comment]
I'd try adding `needs :cxx11` and `ENV.cxx11` and see if that helps.

**Comment by ilovezfs** [comment]
@tseemann :green_apple: because I regenerated the jellyfish-1.1 bottles:

https://github.com/Homebrew/homebrew-science/commit/1805e909cb4ce3cad1b05a0e401a19a1f9c62047

You should now try removing the `need :cxx11` and `ENV.cxx11` and see if we still get :green_apple: or not without them.

**Comment by tseemann** [comment]
@ilovezfs ok, reverting - let's see what happens!

**Comment by tseemann** [comment]
@ilovezfs 
```
brew pull --bottle https://github.com/Homebrew/homebrew-science/pull/2161
==> Fetching patch
Patch: https://github.com/Homebrew/homebrew-science/pull/2161.patch
==> Applying patch
Applying: Kraken 0.10.5-beta (BUG FIX: missing jellyfish-1.1 keg for making databases)
Using index info to reconstruct a base tree...
M	kraken.rb
Falling back to patching base and 3-way merge...
Auto-merging kraken.rb
==> Patch closes issue #2161
Error: Failure while executing: /usr/bin/curl --fail --progress-bar --remote-time --location --user-agent Homebrew/0.9.9\ (Macintosh;\ Intel\ Mac\ OS\ X\ 10.11.6)\ curl/7.43.0 --silent --fail -o /dev/null -I https://github.com/BrewTestBot/homebrew-science/compare/homebrew:master...pr-2161
```

**Comment by tseemann** [comment]
@ilovezfs i can't seem to pull this and commit - i think i need to delete or fix something in the build system?

**Comment by ilovezfs** [comment]
@tseemann there are two problems. The Jenkins build has expired. And this branch needs to be rebased on Homebrew/homebrew-science's master since your fork's master is wildly different.

You can do something like this.
```
brew install hub
hub remote add upstream Homebrew/homebrew-science
hub remote add tseeman
git fetch upstream
git fetch tseeman
git checkout master
git branch -D kraken-jellyfish
git checkout -b kraken-jellyfish tseeman/kraken-jellyfish
git reset --hard upstream/master
git cherry-pick tseeman/kraken-jellyfish
git push -f
```
You'll similarly want to fix your master branch, or just blow your fork away and start over.

**Comment by ilovezfs** [comment]
actually it looks like it's just this branch so you can probably reset to your master and cherry-pick the top commit from the remote branch

**Comment by ilovezfs** [comment]
In other words, assuming origin is your tseeman fork, that will work as shown below
```
Josephs-MacBook-Pro:ts joe$ git clone https://github.com/tseemann/homebrew-science
Cloning into 'homebrew-science'...
remote: Counting objects: 20690, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 20690 (delta 0), reused 0 (delta 0), pack-reused 20687
Receiving objects: 100% (20690/20690), 6.24 MiB | 1.06 MiB/s, done.
Resolving deltas: 100% (12203/12203), done.
Checking connectivity... done.
Josephs-MacBook-Pro:ts joe$ cd homebrew-science
Josephs-MacBook-Pro:homebrew-science joe$ git checkout kraken-jellyfish 
Branch kraken-jellyfish set up to track remote branch kraken-jellyfish from origin.
Switched to a new branch 'kraken-jellyfish'
Josephs-MacBook-Pro:homebrew-science joe$ git reset --hard master
HEAD is now at e40f4c1 spaced: add 20160804 bottle.
Josephs-MacBook-Pro:homebrew-science joe$ git cherry-pick origin/kraken-jellyfish 
[kraken-jellyfish a06b240] Kraken 0.10.5-beta (BUG FIX: missing jellyfish-1.1 keg for making databases)
 Author: Torsten Seemann <torsten.seemann@gmail.com>
 Date: Sun Apr 26 16:59:24 2015 +1000
 1 file changed, 15 insertions(+)
Josephs-MacBook-Pro:homebrew-science joe$ git push -f
```

**Pull request closed without merging**